torchserve/datadog_checks/torchserve/management_api_check.py

```diff
@@ -0,0 +1,184 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import copy
+import json
+import time
+from urllib.parse import urljoin
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.torchserve.model_discovery import ModelDiscovery
+
+WORKER_STATUSES = {
+    "UNKNOWN": 0,
+    "READY": 1,
+    "LOADING": 2,
+    "UNLOADING": 3,
+}
+
+
+class TorchserveManagementAPICheck(AgentCheck):
+    __NAMESPACE__ = 'torchserve.management_api'
+    SERVICE_CHECK_NAME = 'health'
+    CHECK_NAME = "torchserve"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.model_discovery = None
+        self.submit_events = None
+        self.tags = None
+        self.__previous_models = None
+        self.check_initializations.append(self.parse_config)
+
+    def parse_config(self):
+        self.model_discovery = ModelDiscovery(
+            self,
+            limit=self.instance.get("limit", 100),
+            include=self.instance.get("include", [".*"]),
+            exclude=self.instance.get("exclude", []),
+            interval=self.instance.get("interval"),
+        )
+        self.submit_events = self.instance.get("submit_events", True)
+        self.tags = self.instance.get("tags", [])
+        self.tags.append(f"management_api_url:{self.instance['management_api_url']}")
+
+    def check(self, _):
+        for _, _, model, _ in self.model_discovery.get_items():
+            self.process_one_model(model)
+
+        self.service_check(self.SERVICE_CHECK_NAME, self.model_discovery.api_status, tags=self.tags)
+
+        if self.model_discovery.api_status == AgentCheck.OK and self.submit_events and self.model_discovery.refreshed:
+            if self.previous_models is not None:
+                self.compare_models_and_emit_events(self.previous_models, self.model_discovery.all_models)
+
+            self.previous_models = self.model_discovery.all_models
+
+    def process_one_model(self, model):
+        self.log.debug("Processing model [%s]", model['modelName'])
+        model_name = model['modelName']
+        endpoint = f"models/{model_name}/all"
+        url = urljoin(self.instance["management_api_url"], endpoint)
+        response = self.http.get(url)
```

Comment:
Following the other conversation on exception handling, it will be good to move this get request inside try block. If not exceptions other than httperror would go unhandled.

Reply:
addressed